src/library/Navigation/styled.js

```diff
@@ -0,0 +1,169 @@
+import { navigationTheme, navItemTheme } from './themes';
+
+export const NavigationRoot = createStyledComponent(
+  'nav',
+  ({ align, prefix, theme: baseTheme, type }) => {
+    const theme = navigationTheme(baseTheme);
+
+    const aligns = {
+      [ALIGN.start]: 'flex-start',
+      [ALIGN.center]: 'center',
+      [ALIGN.end]: 'flex-end',
+      [ALIGN.justify]: undefined
+    };
+
+    return {
+      backgroundColor: theme[`${prefix}Nav_backgroundColor${type}`],
+      borderBottom: theme[`${prefix}Nav_border`],
+      display: 'flex',
+      justifyContent: aligns[align],
+      listStyle: 'none',
+      margin: 0,
+      // prettier-ignore
+      padding: `${theme[`${prefix}Nav_paddingVertical`]} ${theme[`${prefix}Nav_paddingHorizontal`]}`,
+
+      ...(type === INTERNAL_TYPE.tabs
+        ? {
+            paddingBottom: 0
+          }
+        : undefined),
+
+      '& > :not(:first-child)': {
+        marginLeft: theme[`${prefix}Nav_gutter`]
+      }
+    };
+  },
+  {
+    filterProps: ['prefix'],
+    includeStyleReset: true
+  }
+);
+
+const NavItemButton = createStyledComponent(
+  Button,
+  ({ align, disabled, maxWidth, prefix, selected, theme: baseTheme, type }) => {
+    const theme = navItemTheme(baseTheme);
+
+    return {
+      backgroundColor: disabled && 'transparent',
+      display: 'block',
+
+      '&:hover': {
+        color: !disabled && theme[`${prefix}NavItem_color${type}_hover`]
+      },
+
+      ...(align === ALIGN.justify ? { flexGrow: 1 } : { maxWidth }),
+
+      ...(selected
+        ? {
+            ...(type === INTERNAL_TYPE.tabs
+              ? {
+                  position: 'relative',
+
+                  '&::before': {
+                    // prettier-ignore
+                    backgroundColor:
+                      theme[`${prefix}NavItem_borderColor${type}_selected`],
+                    bottom: -2,
+                    content: '""',
+                    left: -1,
+                    position: 'absolute',
+                    right: -1,
+                    height: 1
+                  }
+                }
+              : {
+                  borderColor:
+                    theme[`${prefix}NavItem_borderColor${type}_selected`]
+                }),
+
+            '&:hover': {
+              backgroundColor:
+                theme[`${prefix}NavItem_backgroundColor${type}_selected`],
+              borderColor:
+                type !== '_tabs'
+                  ? theme[`${prefix}NavItem_borderColor${type}_selected`]
+                  : undefined,
+              color: theme[`${prefix}NavItem_color${type}_selected`]
+            },
+
+            '&, &:focus': {
+              backgroundColor:
+                theme[`${prefix}NavItem_backgroundColor${type}_selected`],
+              color: theme[`${prefix}NavItem_color${type}_selected`]
+            }
+          }
+        : undefined),
+
+      // Truncate
+      '&:active > span > span > span > span > span:focus': {
```

Comment:
Selectors here are definitely a bit fragile.

Reply:
Yep. If we keep using this in many places, always within a Button, we need to just bake it in.